documents/api.py

```diff
@@ -26,12 +27,32 @@ class AttachmentViewSet(ModelViewSet, NestedViewSetMixin):
 
 
 class DocumentViewSet(ModelViewSet):
-    queryset = Document.objects.all()
-    permission_classes = [HasServiceAPIKey | IsAdminUser]
     parser_classes = [MultiPartParser]
     serializer_class = DocumentSerializer
+    # Permission checking is done by the decorators on a method basis
+    permission_classes = [AllowAny]
+
+    def get_queryset(self):
+        if user := self.request.user:
```

Comment:
Minor thing, but I think Django's `AuthenticationMiddleware` will always set at least `AnonymousUser` to request.user so this if-clause is not strictly required here.

Reply:
Nice catch 👍 we might need to add a couple more checks since [django-guardian](https://django-guardian.readthedocs.io/en/stable/configuration.html) apparently uses a different AnonymousUser instance (?) so just to make sure that an actual anonymous user won't be leaking the info from _all_ anonymous users.